### DISCUSSION TITLE
Model a multi-alternative judgement with regression

### DISCUSSION THREAD

**Discussion by hyang336** [lnccbrown/HSSM#284] [Q&A]
I am trying to think through how multiple covariates should be conceptualized in a race_4 model. For example, say I have two covariates, x1 and x2. The research questions is whether one or both of them contribute to the decision process in a 4-alternative forced-choice judgement. 

My initial idea is to use a softmax regression, with the drift-rate v(s) among the 4 accumulators in the race_4 model dependent on linear combinations of x1 and x2. This looks conceptually simple, but I don't know the right syntax to estimate this model in HSSM. In particular, what should the formula and link function be? Is there a softmax link function in Bambi?

Thanks,
HY

**Comment by AlexanderFengler** [comment]
Hey HY, 

As a first pass, I would suggest to model v1, v2, v3, v4 as separate regressions (the tutorials should provide you with sufficient info for that scenario). 

The soft-max regression approach is interesting, but I think we will have trouble shoe-horning that into the basic HSSM workflow with simple reliance on Bambi. 

You will be able to build such a model as a custom pymc model, using the HSSM provided random variables (in your case the race_4 model). 

Best,
Alex

**Reply by hyang336** [reply]
Hi, Alex,

Thanks for the suggestions. I actually realized last night that the softmax regression wouldn't be suitable for my hypothesis. 

In our task we have subjects make confidence ratings while being scanned. The response is presumably made on a 1-d signal (e.g. confidence), and the covariate is a 1-d signal (e.g. trial-level BOLD). The hypothesis is that intermediate level of neural activity will provide strong evidence for the intermediate level of confidence. So, if we have one accumulator for each response option, the intermediate accumulator should have the highest drift rate when the neural activity is intermediate as well. In other words, drift-rate(v)=f(neural-data(x)), where f(x) needs to be able to express monotonic and nonmonotonic relationship between x and v. Inspired by the beta distribution, I'm thinking of something like this for regression for each v: v=ln(f(x,a,b))=ln(C) + (a-1)ln(x) + (b-1)ln(1-x), x in [0,1]. 

Because I have never used beta distribution before (my stats knowledge is mostly with normal distribution), I am not really sure if this is the correct approach, or if it is the most efficient approach. Your feedback will be highly appreciated.

Thanks,
HY

**Comment by AlexanderFengler** [comment]
Hi HY, 

Bambi will generally follow this logic:

You have some linear model `y = a + bx1 + cx2 .....`, then you may want to apply a link function,
`f(y)`, where you have quite some flexibility in terms of the link you choose (custom ones are allowed, maybe check the bambi documentation to see in how far that can help). Your model parameter that you hooked up with the linear model will then take the form e.g.: `v_1 = f(y)`.

---

So for a fixed functional form `f()`, you should be able to use this mechanism (your link function could take the form of a beta distribution e.g.). However note that the definition of *link* function is in fact the **inverse** of the forward transform, see also [here](https://en.wikipedia.org/wiki/Generalized_linear_model)).

So you could test a "`beta-link`" again a "`monotone-link`" e.g.. 

Best,
Alex


However where things become a little more tricky is if you actually want to treat the parameters `a`, `b` in your function `f(x, a, b)` as random variables as well (fit them / sample from them). So in other words apply parameterized link functions. That, to the best of my knowledge, will not work with Bambi (yet). However again, you can build a custom pymc model with HSSM random variables to accomodate this too.

**Reply by hyang336** [reply]
Can I simply squash my covariates (x) to [0,1] and then log-transform it first? Wouldn't this make the model just a simple linear model with an identity link function in the form v_1 = a + b ln(x) + c ln(1-x) ? In this case Bambi will be able to estimate a, b, and c as in any GLM. Or am I missing something?

Thanks,
HY

**Reply by hyang336** [reply]
Well I tried this using the race_no_bias_angle_4 model and parameter recovery was pretty bad, could be due to the priors on those parameters or the chain not being long enough (500 burn 500 sample). Or is this approach fundamentally wrong?
